File: binary search.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1

# User input
n = int(input("Enter number of elements: "))

arr = []
print("Enter sorted elements:")
for i in range(n):
    arr.append(int(input()))

target = int(input("Enter element to search: "))

result = binary_search(arr, target)

if result != -1:
    print("Element found at index", result)
else:
    print("Element not found")

File: BinarySearch.java
public class BinarySearch {

    public static void main(String[] args) {
        int arr[] = {1,2,3,4,5,6,7};
        int target=3;
        System.out.println(search(arr, target));
    }

    public static int search(int arr[], int target) {
       int first = 0;
       int last = arr.length - 1;

       while (first <= last) {
           int mid = (first + last) / 2;
           if (arr[mid] == target) {
               return mid;
           }
           if (target > arr[mid]) {
               first = mid + 1;
           } else {
               last = mid - 1;
           }
       }
       return -1;
    }
}